rfcs/0000-tsdb-dimensions.md

```diff
@@ -0,0 +1,240 @@
+-->
+
+## Usage
+
+<!--
+Stage 1: Describe at a high-level how these field changes will be used in practice. Real world examples are encouraged. The goal here is to understand how people would leverage these fields to gain insights or solve problems. ~1-3 paragraphs.
+-->
+
+Integration package development is the key beneficiary of this change. The fields of the document that are received from an integration receives a field mapping. If and when TSDB benefits are to be utilised, along with the field mapping with a metric type, atleast one of the fields  must recieve `dimension: true` annotation. 
+
+Example of field mapping in integrations with the field enabled as a dimension field.
+```yaml
+---
+- name: wait_class
+  type: keyword
+  description: Every wait event belongs to a class of wait event. 
+  dimension: true
+
+```
+## Source data
+
+The source of this data comes from monitoring a host like a Linux machine, laptop or a k8s node. The can come delivered through different shippers like Elastic Agent system metrics inputs, apm agents, prometheus node exporter and other host metric collectors.
+<!--
+Stage 1: Provide a high-level description of example sources of data. This does not yet need to be a concrete example of a source document, but instead can simply describe a potential source (e.g. nginx access log). This will ultimately be fleshed out to include literal source examples in a future stage. The goal here is to identify practical sources for these fields in the real world. ~1-3 sentences or unordered list.
+-->
+
+<!--
+Stage 2: Included a real world example source document. Ideally this example comes from the source(s) identified in stage 1. If not, it should replace them. The goal here is to validate the utility of these field changes in the context of a real world example. Format with the source name as a ### header and the example document in a GitHub code block with json formatting, or if on the larger side, add them to the corresponding RFC folder.
+-->
+
+<!--
+Stage 3: Add more real world example source documents so we have at least 2 total, but ideally 3. Format as described in stage 2.
+-->
+
+## Scope of impact
+
+<!--
+Stage 2: Identifies scope of impact of changes. Are breaking changes required? Should deprecation strategies be adopted? Will significant refactoring be involved? Break the impact down into:
+ * Ingestion mechanisms (e.g. beats/logstash)
+ * Usage mechanisms (e.g. Kibana applications, detections)
+ * ECS project (e.g. docs, tooling)
+The goal here is to research and understand the impact of these changes on users in the community and development teams across Elastic. 2-5 sentences each.
+-->
+
+## Concerns
+
+No concerns are known as of now. Presence of the `dimension:true` does not impact functionality. Elastic Stack version 8.7 is essential for this. 
```

Comment:
If 8.7+ is required to support `dimension` and the proposed changes here were added to ECS, would any integration package using the version of ECS including `dimension` need to explicitly support 8.7+ going forward?

Reply:
No, any package using the proposed ECS version can continue to serve Non-TSDB and TSDB settings because `dimension: true` supports both Non-TSDB and TSDB settings .

 A package can upgrade to use the proposed ECS version (with` dimension: true`) and continue to use a version prior to 8.7.

Reply:
@agithomas Are you saying that unless we require that the package minimum version is 8.7, we cannot update our ECS versions?
At SEI we upgrade ECS versions for all our integrations every release, however we do not bump the minimum version for our packages..

Reply:
@P1llus , If only the package has to be TSDB enabled, proposed ECS (having `dimension : true`) is to be used. 

If the package do not want to have TSDB enablement, it can continue with old version of ECS.

Reply:
That is the problem though @agithomas . That means any package that wants to have other ECS changes other than `dimension: true` has to upgrade their minimum version.

Since we always update our ECS version of packages, we would lock users out of hotfixes for older versions.

While we had planned to bump the minimum version of packages at some point, this change would force that at this point.

Reply:
> That is the problem though @agithomas . That means any package that wants to have other ECS changes other than dimension: true has to upgrade their minimum version.

No. Even if you run Elasticsearch 8.3 and have `dimension: true` set, it will work as the field is just ignored. You just keep having the old behaviour.